presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -596,30 +599,42 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
         RaptorTableHandle handle = checkType(tableHandle, RaptorTableHandle.class, "tableHandle");
         long tableId = handle.getTableId();
 
-        ImmutableList.Builder<RaptorColumnHandle> columnHandles = ImmutableList.builder();
+        ImmutableList.Builder<RaptorColumnHandle> columnHandlesBuilder = ImmutableList.builder();
         ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
         for (TableColumn column : dao.getTableColumns(tableId)) {
-            columnHandles.add(new RaptorColumnHandle(connectorId, column.getColumnName(), column.getColumnId(), column.getDataType()));
+            columnHandlesBuilder.add(new RaptorColumnHandle(connectorId, column.getColumnName(), column.getColumnId(), column.getDataType()));
             columnTypes.add(column.getDataType());
         }
 
+        List<RaptorColumnHandle> columnHandles = columnHandlesBuilder.build();
         long transactionId = shardManager.beginTransaction();
 
         setTransactionId(transactionId);
 
         Optional<String> externalBatchId = getExternalBatchId(session);
         List<RaptorColumnHandle> sortColumnHandles = getSortColumnHandles(tableId);
         List<RaptorColumnHandle> bucketColumnHandles = getBucketColumnHandles(tableId);
+
+        Optional<RaptorColumnHandle> temporalColumnHandle = empty();
```

Comment:
I think this can be done like
```java
Optional<RaptorColumnHandle> temporalColumnHandle = Optional.ofNullable(dao.getTemporalColumnId(tableId))
        .map(...)
```